pkg/csi/cinder/nodeserver.go

```diff
@@ -448,13 +372,15 @@ func (ns *nodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 	if len(volumePath) == 0 {
 		return nil, status.Error(codes.InvalidArgument, "Volume path not provided")
 	}
+	volCapability := req.GetVolumeCapability()
+	if volCapability != nil && volCapability.GetBlock() != nil {
+		// The Node plugin only resizes mounted volumes, so we need to skip block unmanaged volumes.
+		return &csi.NodeExpandVolumeResponse{}, nil
+	}
```

Comment:
what makes you think that `GetBlock() != nil` means that the volume is unmanaged? For me it looks like an unrelated bug fix. And to be honest I don't understand the bug description. For me it looks like this change breaks volume resize behavior.
https://github.com/kubernetes-csi/external-resizer/blob/f387899b5a5af4e601a93f1c963aa646f4e54c1b/pkg/resizer/csi_resizer.go#L208-L229 - the `VolumeCapability` is always not nil for block devices.

Reply:
Hi, the resizer only works with the CSI-controller.

The CSI-node runs `resize2fs` or similar tools for formatted (managed) disks. The pod takes care of managing the block device

Reply:
> Hi, the resizer only works with the CSI-controller.

right, the controller generates the `NodeExpandVolume` call, which is then handled by the node-driver. Anyway, I took a look at the AWS CSI controller and it has the same logic. I was confused about the block type. I associated block type with all cinder volumes and fstype with manila shared filesystem storage. Now I see that block type is a raw volume without specified fs.

Anyway, your fix needs to be implemented in a separate PR, since there is a case, when you have a raw volume being resized, and some hypervisors require a block device rescan to notify the guest OS with a new block device size. And since the `RescanBlockDeviceGeometry` func relies on the `volumePath`, there must be a new func that just rescans the block device without checking the FS size.

Reply:
Ok, i've remove it.

Reply:
@sergelogvinov can you open a new issue explaining this use case?